packages/material-ui/src/ButtonBase/ButtonBase.d.ts

```diff
@@ -72,11 +72,10 @@ export interface ButtonBaseTypeMap<P = {}, D extends React.ElementType = 'button
      * The system prop that allows defining system overrides as well as additional CSS styles.
      */
     sx?: SxProps<Theme>;
-    // @types/react is stricter
     /**
      * @default 0
      */
-    tabIndex?: string | number;
+    tabIndex?: number;
```

Comment:
To ensure future compatibility let's reuse the same types with something like `NonNullable<React.HTMLAttributes<any>['tabIndex']>`

Reply:
Sure. Why `NonNullable`?

Reply:
`tabIndex?: React.HTMLAttributes<any>['tabIndex']` would be `tabIndex?: number | undefined` while `tabIndex?: NonNullable<React.HTMLAttributes<any>['tabIndex']>` would be `tabIndex?: number`. These types are slightly different.

Reply:
Thanks for the explanation. Pushed the requested change.